.github/workflows/publish_website.yml

```diff
@@ -0,0 +1,31 @@
+# Builds the website sourced from docs/website/ using `mkdocs` and pushes
+# to the gh-pages branch for publishing on GitHub Pages.
+#
+# See https://squidfunk.github.io/mkdocs-material/publishing-your-site/
+
+name: Publish Website
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+defaults:
+  run:
+    working-directory: docs/website
+
+jobs:
+  publish_website:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout out repository
+        uses: actions/checkout@v2
+      - name: Setting up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+      - name: Installing Material for MkDocs
+        run: pip install mkdocs-material
+      - name: Deploying to gh-pages
+        run: mkdocs gh-deploy --force
```

Comment:
Yeah just tested locally. It looks like it just needs the previous gh-pages history and then it's fine. We can fetch that as well. I'll send a PR to your PR :-D

Reply:
https://github.com/ScottTodd/iree/pull/2